dummyfs/dummyfs.c

```diff
@@ -876,8 +958,34 @@ int dummyfs_mount(void **ctx, const char *data, unsigned long mode, oid_t *root)
 	o->oid.port = fs->port;
 	o->mode = S_IFDIR | 0666;
 
-	memcpy(&rootdir, &o->oid, sizeof(oid_t));
+	rootdir = o->oid;
+	fs->parent = rootdir;
+
+	if (data != NULL) {
+		abspath = resolve_path(data, NULL, 1, 0);
+		if (lookup(abspath, &fs->parent, NULL) < EOK) {
+			free(abspath);
+			dev_cleanup(fs);
+			object_cleanup(fs);
+			resourceDestroy(fs->mutex);
+			free(fs);
+			return -ENOENT;
+		}
+
+		fs->mountpt = strdup(abspath);
+		free(abspath);
```

Comment:
I guess you want to conserve memory (as `resolve_path` allocated `PATH_MAX`), comment would be good here to deter from optimising this code to `fs->mountpt = abspath;`

Reply:
Hmm, I think `realloc(abspath, strlen(abspath) + 1);` would also do the trick and would be pretty self-commenting.